alica_engine/src/engine/RuleBook.cpp

```diff
@@ -435,7 +435,7 @@ PlanChange RuleBook::transitionRule(RunningPlan& r)
         return PlanChange::NoChange;
     const State* nextState = nullptr;
 
-    if (!r.getBasicPlan() || !r.getBasicPlan()->initExecuted()) {
+    if (!r.getBasicPlan()) {
```

Comment:
I don't think so.  there is a check above :
if (r.getActiveState() == nullptr)

and there shouldn't be any active state unless the assignment succeed. Guess the issue you fixed here was primarily because a state was assigned by alica thread but its init wasn't executed by the scheduler thread. Let me know if that wasn't the case.

Reply:
the transition rule is calling moveState which sets the active state to a valid value so it won't be nullptr I think. Also I thought that alica always evaluated the transitions first in order to avoid unnecessary activation of any plans/behaviours in a state. We even relied on this behaviour especially in pause/resume scenarios

Reply:
the transition rule is calling moveState which sets the active state to a valid value so it won't be nullptr I think. Also I thought that alica always evaluated the transitions first in order to avoid unnecessary activation of any plans/behaviours in a state. We even relied on this behaviour especially in pause/resume scenarios

Reply:
my bad, I think I may have mistaken the code here. You are right, it shouldn't be required.